### PR TITLE
statistics: fix row count when not fully loading ver2 stats

### LINKED ISSUE #25387: row count is wrong when using v2 stats and there's only topn and not fully load stats
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
In `columnCountFromStorage()`:
https://github.com/pingcap/tidb/blob/a45b3aaa2b1ac55bcce8d5a5d0018c9f56bc3f06/statistics/handle/handle.go#L1114-L1143
We return directly if `select sum(count) from mysql.stats_buckets` returns NULL.
It will return NULL if there aren't any buckets, but there may still exist topn, so we shouldn't return directly here.

### 2. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #25387

### What is changed and how it works?

Please see the issue for details.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- no release note.

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -1112,17 +1112,18 @@ func (h *Handle) histogramFromStorage(reader *statsReader, tableID int64, colID
...
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
-	if rows[0].IsNull(0) {
-		return 0, nil
-	}
-	count, err := rows[0].GetMyDecimal(0).ToInt()
-	if err != nil {
-		return 0, errors.Trace(err)
+	if !rows[0].IsNull(0) {
```
Comment (winoros) [inline statistics/handle/handle.go:1120]: We may need to add some comments. From the codes, we can see that we've modified the codes to be adjusted with Version2 but still found something wrong this time.

**Comment by winoros** [comment]
No need to pick to 5.0, removed the label

File: statistics/handle/handle.go (hunk 1 above)
Comment (time-and-fate) [reply statistics/handle/handle.go:1121]: Added some comments. PTAL again.

**Review by winoros — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**